federation/pkg/federation-controller/service/endpoint_helper.go

```diff
@@ -31,22 +31,42 @@ import (
 // worker runs a worker thread that just dequeues items, processes them, and marks them done.
 // It enforces that the syncHandler is never invoked concurrently with the same key.
 func (sc *ServiceController) clusterEndpointWorker() {
-	fedClient := sc.federationClient
+	// process all pending events in endpointWorkerDoneChan
+	eventPending := true
```

Comment:
the break statement in default case will only break the select loop, and the goroutine will get stuck in for loop forever, so we need this variable to come out of for loop